File: src/parser/fixtures/java/UserService.java
package com.example.demo.service;

import com.example.demo.domain.User;
import com.example.demo.repository.UserRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class UserService extends BaseService implements UserOperations {

    private final UserRepository userRepository;
    private final Notifier notifier;

    public UserService(UserRepository userRepository, Optional<Notifier> notifier) {
        this.userRepository = userRepository;
        this.notifier = notifier.orElse(null);
    }

    public List<User> findUsers() {
        return userRepository.findAll();
    }
}

File: src/parser/fixtures/kotlin/UserService.kt
package com.example.demo.service

import com.example.demo.domain.User
import com.example.demo.repository.UserRepository
import org.springframework.stereotype.Service

@Service
class UserService(
    private val userRepository: UserRepository,
    private val notifier: Notifier?
) : BaseService(), UserOperations {

    fun findUsers(): List<User> {
        return userRepository.findAll()
    }
}
